phpdotnet/phd/Package/PHP/Web.php

```diff
@@ -224,6 +224,28 @@ protected function writeJsonIndex() {
         $ids = array();
         $desc = array();
         foreach($this->indexes as $id => $index) {
+            // Skip indexes that are not chunks
+            if (!$index["chunk"]) {
+                continue;
+            }
+
+            // Handle items with long description but no short description
+            if (empty($index["sdesc"]) && !empty($index["ldesc"])) {
```

Comment:
What is the purpose of `empty` here? Is it really possible that the key is undefined? The existing code below seems to suggest that this cannot be the case.

Reply:
Thanks for the quick review.

Looks like both `sdesc` and `ldesc` are initialized as empty strings:
https://github.com/php/phd/blob/417b06655c89b5bde5794dee58ce4e41834c6ef0/phpdotnet/phd/Index.php#L209-L213

However, similar `empty()` checks exist elsewhere:
https://github.com/php/phd/blob/417b06655c89b5bde5794dee58ce4e41834c6ef0/phpdotnet/phd/Index.php#L360
https://github.com/php/phd/blob/417b06655c89b5bde5794dee58ce4e41834c6ef0/phpdotnet/phd/Index.php#L372
https://github.com/php/phd/blob/417b06655c89b5bde5794dee58ce4e41834c6ef0/phpdotnet/phd/Package/PEAR/Web.php#L73-L77

So I decided to just play it safe and use `empty` as well.

For consistency, I suggest opening a separate issue to review these checks project-wide (it's hacktoberfest!). I can also address just this instance if preferred, it shouldn't break anything. What are your thoughts on the best approach?

Reply:
For consistency with the current function, let's not use `empty` here. Yes, the rest can be refactored at a later point of time.

Reply:
Done, and issue created #158